importers/eag_importer.py

```diff
@@ -0,0 +1,205 @@
+            platform = data_store.get_platform(platform_name=callsign, change_id=change_id,)
+
+            # Sensor Type is always fixed to GPS
+            sensor_type = data_store.add_to_sensor_types("GPS", change_id=change_id).name
+            sensor = platform.get_sensor(
+                data_store=data_store, sensor_type=sensor_type, change_id=change_id,
+            )
+            state = datafile.create_state(data_store, platform, sensor, timestamp, self.short_name)
+
+            # Location and height info are provided as Earth-Centred, Earth-Fixed co-ordinates
+            # We convert them to lat/lon and height
+            location, height, success = self.convert_ecef_to_location_and_height(
+                ecef_x_token, ecef_y_token, ecef_z_token
+            )
+
+            if success:
+                state.location = location
+                combine_tokens(ecef_x_token, ecef_y_token, ecef_z_token).record(
+                    self.name, "location", state.location, "ECEF X, Y and Z"
+                )
+
+                state.elevation = height * unit_registry.metre
+                combine_tokens(ecef_x_token, ecef_y_token, ecef_z_token).record(
+                    self.name, "altitude", state.elevation, "ECEF X, Y and Z"
+                )
+            else:
+                self.errors.append(
+                    {
+                        self.error_type: f"Error on line {line_number}. Cannot parse ECEF X, Y and Z to latitude, longitude and height: {line}"
+                    }
+                )
+                continue
+
+            heading_valid, heading = convert_absolute_angle(
+                heading_token.text, line_number, self.errors, self.error_type
+            )
+            if heading_valid:
+                state.heading = heading
+                heading_token.record(self.name, "heading", heading)
+
+    def get_previous_sunday(self, date_of_recording_str):
+        format_str = "%Y%m%d"
+        date_of_recording = datetime.datetime.strptime(date_of_recording_str, format_str)
+
+        sun_offset = (date_of_recording.weekday() - 6) % 7
+
+        last_sun = date_of_recording - datetime.timedelta(days=sun_offset)
+
+        return last_sun
+
+    def calculate_timestamp(self, time_since_sun_ms, last_sun_date):
+        timestamp = last_sun_date + datetime.timedelta(milliseconds=time_since_sun_ms)
+
+        timestamp = timestamp + datetime.timedelta(seconds=self.TIME_OFFSET)
+
+        return timestamp
+
+    def convert_ecef_to_location_and_height(self, ecef_x_token, ecef_y_token, ecef_z_token):
+        """Converts Earth-Centred, Earth-Fixed X, Y and Z co-ordinates into a Location object containing
+        latitude and longitude, plus a height value in metres.
+
+        X, Y and Z should be provided in metres.
+        
+        All formulae taken from https://microem.ru/files/2012/08/GPS.G1-X-00006.pdf
+        """
+        ecef_x = float(ecef_x_token.text)
```

Comment:
Is it possible that any of X, Y, or Z coordinates are wrong? I don't know how robust it this file but float conversion might fail if any of them is a not convertible string.

Reply:
You're right - I thought I had a try-catch block around that, but I didn't. I'll fix it now.

Reply:
Fixed now in 58cfa2b